.github/workflows/pulser-setup/action.yml

```diff
@@ -0,0 +1,31 @@
+name: Pulser setup
+description: "Sets up Python and installs Pulser."
+inputs:
+  python-version:
+    description: Python version
+    required: false
+    default: '3.9'
+  extra-packages:
+    description: Extra packages to install (give to grep)
+    required: false
+    default: ''
+runs:
+  using: 'composite'
+  steps: 
+    - name: Set up Python
+      uses: actions/setup-python@v4
+      with:
+        python-version: ${{ inputs.python-version }}
+        cache: 'pip'
+    - name: Install Pulser
+      shell: bash
+      run: |
+        python -m pip install --upgrade pip
+        pip install -e ./pulser-core -e ./pulser-simulation
```

Comment:
I didn't think so either and I tried to do it without it, but then actually `pytest-cov` failed because the source code was located in `site-packages` rather than in the root of the repo.